cognite/client/data_classes/time_series.py

```diff
@@ -86,6 +91,54 @@ def __init__(
             )
         self.legacy_name = legacy_name
 
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        """Dump the object to a dictionary"""
+        output: dict[str, Any] = {}
+        if self.external_id is not None:
+            output["externalId" if camel_case else "external_id"] = self.external_id
+        if self.instance_id is not None:
+            output["instanceId" if camel_case else "instance_id"] = self.instance_id.dump(
+                camel_case=camel_case, include_instance_type=False
+            )
```

Comment:
Need manual implementation of load/dump methods now as instance id introduces a nested structure.

Reply:
The default implementation only accepts basic basic types not nested. The moment you need nested you need to implement load/dump. It is possible to write this

```
    def dump(self, camel_case: bool = True) -> dict[str, Any]:
        """Dump the object to a dictionary"""
        output = super().dump(camel_case=camel_case)
        if self.instance_id is not None:
            output["instanceId" if camel_case else "instance_id"] = self.instance_id.dump(
                camel_case=camel_case, include_instance_type=False
            )
        return output
```

but I have a preference for the explicit method here.